### PR TITLE
cnao, Add unit test job for release and stable jobs

### PR DESCRIPTION
**Pull request by RamLavi**
CNAO is moving from travis ci to a prow unit-test job.

### CODE REVIEW

**Comment by RamLavi** [comment]
/hold

should be merged only after the script is merged on CNAO: [PR](https://github.com/kubevirt/cluster-network-addons-operator/pull/729)

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:85]: Do you need so much memory for unit tests? I guess you didn't have that much in travis, maybe better start with less (4Gi?) and adapt if needed, wdyt?

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:71]: I think we don't need so many concurrent tests running in parallel, maybe better remove this setting.
```suggestion
```

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:90]: docker in docker requires privileged mode.
```suggestion
              - "automation/check-patch.unit-test.sh"
            securityContext:
              privileged: true
```

**Comment by RamLavi** [comment]
hey @fgimenez, looks like `prow-deploy-test ` lane fails:
```
Stopping Docker: dockerProgram process in pidfile '/var/run/docker-ssd.pid', 1 process(es), refused to die.
```

do you think it's related to something I did in the PR?

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.44.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.44.yaml:71]:
```suggestion
```

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.44.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.44.yaml:90]:
```suggestion
              - "automation/check-patch.unit-test.sh"
            securityContext:
              privileged: true
```

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits.yaml:171]:
```suggestion
```

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits.yaml:190]:
```suggestion
              - "automation/check-patch.unit-test.sh"
            securityContext:
              privileged: true
```

**Review by fgimenez — CHANGES_REQUESTED** [review]

**Comment by fgimenez** [comment]
> do you think it's related to something I did in the PR?

@RamLavi i don't think so, from https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/811/prow-deploy-test/1346802302165454848 the actual error is:
```
"./cluster-up/up.sh", "unknown flag: --run-etcd-on-memory
```
Will take a look, feel free to ignore.

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (RamLavi) [reply github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:85]: I agree 4Gi would be more than enough.

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (RamLavi) [reply github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:90]: I think the container is already configured to privileged mode [here](https://github.com/kubevirt/project-infra/pull/811/files#diff-61d6d3234278918d4dbac6e78ff300a63d0c6a493bdd75b1797d39a6ef790cfcR82). do I need to add the privilege configuration somewhere else?

**Comment by fgimenez** [comment]
> Will take a look, feel free to ignore.

prow-deploy-test failure should be fixed with https://github.com/kubevirt/project-infra/pull/813 can you pls rebase on top of current master? that should make the test pass

File: github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml
Comment (fgimenez) [reply github/ci/prow/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-presubmits-0.42.yaml:89]: Indeed that's perfect, sorry, I missed it

**Comment by RamLavi** [comment]
Hey @fgimenez, please see review fixes [here](https://github.com/kubevirt/project-infra/compare/a9738126712f3e94aded6d1fb09c08a0a55d83ca..284db808a2622abae90347035f720645f0cf2dd5)

**Comment by RamLavi** [comment]
> > Will take a look, feel free to ignore.
> 
> prow-deploy-test failure should be fixed with #813 can you pls rebase on top of current master? that should make the test pass

Done

**Comment by RamLavi** [comment]
The test is failing since the CNAO PR is not merged yet. will unhold and re-try

**Comment by fgimenez** [comment]
> The test is failing since the CNAO PR is not merged yet. will unhold and re-try

ok, we could also try the test from your branch before merging adding an `extra_refs` property, if you have already tested it all should be ok.

**Comment by RamLavi** [comment]
> > The test is failing since the CNAO PR is not merged yet. will unhold and re-try
> 
> ok, we could also try the test from your branch before merging adding an `extra_refs` property, if you have already tested it all should be ok.

nah I already merged it, let's try it here.

/rehearse

**Comment by RamLavi** [comment]
looks like it fails running [this](https://github.com/kubevirt/cluster-network-addons-operator/blob/679d2952daaa262224e93be86301f52ab4e695b0/build/registry/Dockerfile#L7) command in the docker file. 
does it require some special privileges?

**Comment by fgimenez** [comment]
> does it require some special privileges?

hm this is weird, i don't think so, that should work afaik. building locally the image works, right?

i see an additional error at the end, on line 8581:
```
+ cp /logs/artifacts
cp: missing destination file operand after '/logs/artifacts'
Try 'cp --help' for more information.
```
like if `$ARTIFACTS` would not be defined here https://github.com/kubevirt/cluster-network-addons-operator/blob/master/automation/check-patch.unit-test.sh#L8 ?

**Comment by RamLavi** [comment]
> > does it require some special privileges?
> 
> hm this is weird, i don't think so, that should work afaik. building locally the image works, right?
> 
> i see an additional error at the end, on line 8581:
> 
> ```
> + cp /logs/artifacts
> cp: missing destination file operand after '/logs/artifacts'
> Try 'cp --help' for more information.
> ```
> 
> like if `$ARTIFACTS` would not be defined here https://github.com/kubevirt/cluster-network-addons-operator/blob/master/automation/check-patch.unit-test.sh#L8 ?

you're right, that's the issue. will fix on CNAO and recheck here

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
